readme.md

```diff
@@ -282,6 +294,57 @@ When given an unsupported protocol.
 
 The promise returned by Got has a `.cancel()` function which, when called, aborts the request.
 
+## Cache
+
+You can use the JavaScript `Map` type as an in memory cache:
+
+```js
+const got = require('got');
+const map = new Map();
+
+(async () => {
+    let response = await got('todomvc.com', {cache: map});
+    console.log(response.fromCache);
```

Comment:
`. fromCache` and any other extra properties on the response needs to be documented.